src/main/java/org/folio/helper/CheckinHelper.java

```diff
@@ -358,6 +359,9 @@ private Future<Void> checkinItem(JsonObject itemRecord, CheckInPiece checkinPiec
     if (StringUtils.isNotEmpty(checkinPiece.getBarcode())) {
       itemRecord.put(ITEM_BARCODE, checkinPiece.getBarcode());
     }
+    if (StringUtils.isNotEmpty(checkinPiece.getAccessionNumber())) {
+      itemRecord.put(ITEM_ACCESSION_NUMBER, checkinPiece.getAccessionNumber());
+    }
```

Comment:
As I remember we have couple of places where we do this converting, checking

Reply:
We have similar conversion in ReceivingHelper as well

Reply:
I've noticed that the CheckinHelper is utilized when "receiving" items, whereas the ReceivingHelper is employed when performing an "unreceive." Since we do not make any changes to `accessionNumber` during an "unreceive" operation, I haven't implemented any changes in that context. However, considering that the inner workings are not entirely clear to me, I'm more than willing to apply the modifications to the ReceivingHelper as well, if that's deemed necessary.

Reply:
okay I see that CheckingHelper and ReceivingHelper differs on ITEM_DISCOVERY_SUPPRESS, so they already are not strong consistent, I am ok to leave as it if you thinking that only CheckingHelper requires this field